ray_on_golem/server/services/ray.py

```diff
@@ -216,6 +228,35 @@ def _get_new_node_id(self) -> NodeId:
         self._nodes_id_counter += 1
         return node_id
 
+    async def _monitor_node(self, node_id: NodeId, activity: Activity) -> None:
+        async with self._get_node_context(node_id) as node:  # type: Node
+            is_head_node = self._is_head_node(node.tags)
+        while True:
+            try:
+                activity_state = await activity.get_state()
```

Comment:
I wouldn't assume the activity failed until we retry this and e.g. three consecutive calls to `get_state` fail...

Reply:
unless of course, the function _does_ return with a state of e.g. `Terminated`  from which there's no return ;)

Reply:
get_state has 2 retries(via decorator), I can make it 3(or 5?)

Reply:
hmm... I'm not fond of that pattern... it should be up to the user of the library to decide how many and if any retries should be made... of course, there can be a default but it should be overrideable from user code...